### ISSUE TITLE
cannot reproduce fig3

### ISSUE THREAD

**Issue by zym1010** [david-klindt/NIPS2017#4]
Hi, 

I'm trying to do reproduce fig 3 using your code, with Tensorflow 1.4.1 and Python 3.6. However, there are several problems.

1. `fig3.ipynb` is not generated using the same `training.py` as on the GitHub. For example, in the GitHub version of `fig3.ipynb`, the return value of `training.train` has 10 elements. However, in the GitHub version of `training.py`, there are 13 elements on return. There are also other differences, such as out of range error for dropout. etc #3 
2. Even with all these fixed, I cannot get the results on GitHub. attached is my result of running the code. The train/val curves look very different than those on GitHub. any idea why?

<img width="1130" alt="screen shot 2018-02-04 at 11 30 57 am" src="https://user-images.githubusercontent.com/4273204/35779718-be7763c4-099f-11e8-87c8-d7207498e204.png">

**Comment by zym1010** [comment]
any update?

**Comment by david-klindt** [comment]
Hi, sorry for the delay. Yes you are right, as pointed out in [#3](https://github.com/david-klindt/NIPS2017/issues/3), the code grew somewhat organically over the figures and I missed updating it. I did that now and it should work (see commit). Please let me know if you are still experiencing troubles!
